pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -109,9 +109,48 @@ func (p *staticPolicy) Name() string {
 }
 
 func (p *staticPolicy) Start(s state.State) {
-	// Configure the shared pool to include all detected CPU IDs.
-	allCPUs := p.topology.CPUDetails.CPUs()
-	s.SetDefaultCPUSet(allCPUs)
+	// flag to indicate whether ignore state from file and start fresh state
+	initializeNewEmptyState := false
+
+	// static policy must have at least one cpu in defaultCPUSet
+	if s.GetDefaultCPUSet().IsEmpty() {
+		// State is initialized empty
+		// Configure the shared pool to include all detected CPU IDs.
+		initializeNewEmptyState = true
+	} else {
+		// State has already been initialized
+
+		// 1 Reserved cpuset is not part of default cpuset because:
+		// - kube/system reserved have changed (increased) - may lead to some containers not being able to start
+		// - user tampered with file
+		if !p.reserved.Intersection(s.GetDefaultCPUSet()).Equals(p.reserved) {
```

Comment:
Are these heuristics going to be reliable? I’m concerned about the case where the policy flag changes across kubelet restart. Could we add the policy name to the checkpoint file? Sorry, should have thought of that during the last review.